xscen/diagnostics.py

```diff
@@ -97,535 +110,256 @@ def _invert_unphysical_temperatures(
-#         num = [num[dim] for dim in mask.dims]
-#
-#     slices = [
-#         [slice(i * (L // N), (i + 1) * (L // N)) for i in range(N)]
-#         for L, N in zip(mask.shape, num)
-#     ]
-#
-#     points = []
-#     for zone_slices in itertools.product(*slices):
-#         zone = mask.isel({dim: slic for dim, slic in zip(mask.dims, zone_slices)})
-#         points.append(random_site_selection(zone, 1))
-#
-#     return xr.concat(points, 'site')
-
-
-def fig_compare_and_diff(
-    sim: xr.DataArray, scen: xr.DataArray, op: str = "difference", title: str = ""
-) -> plt.figure:
-    """Plot sim and scen, as well the their difference, improvement or distance.
+# TODO: just measures?
+
+
+def properties_and_measures(
+    ds: xr.Dataset,
+    properties: Union[
+        str, PosixPath, Sequence[Indicator], Sequence[Tuple[str, Indicator]], ModuleType
+    ],
+    period: list = None,
+    unstack: bool = False,
+    dref_for_measure: Optional[xr.Dataset] = None,
+    unit_conversion: Optional[dict] = None,
+    to_level_prop: str = "diag-properties",
+    to_level_meas: str = "diag-measures",
+):
+    """
+    Calculate properties and measures of a dataset.
 
     Parameters
     ----------
-    sim: xr.DataArray
-        First data to plot (normally an xclim.sbda.properties of an unajusted simulation, but could be anything)
-    scen: xr.DataArray
-        Second data to plot (normally an xclim.sbda.properties of an ajusted simulation, but could be anything)
-    op: str
-        Operation for the third panel
-        If 'distance': abs(sim - scen)
-        If 'improvement': abs(sim) - abs(scen)
-        If 'difference': sim - scen
-    title: str
-        Title of the figure
+    ds: xr.Dataset
+        Input dataset.
+    properties: Union[str, PosixPath, Sequence[Indicator], Sequence[Tuple[str, Indicator]]]
+      Path to a YAML file that instructs on how to calculate properties.
+      Can be the indicator module directly, or a sequence of indicators or a sequence of
+      tuples (indicator name, indicator) as returned by `iter_indicators()`.
+    period: lst
+        [start, end] of the period to be evaluated. The period will be selected on ds
+        and dref_for_measure it it is given.
+    unstack: bool
+        Whether to unstack ds before computing the properties.
+    dref_for_measure: xr.Dataset
+        Dataset of properties to be used as the ref argument in the computation of the measure.
+        Ideally, this is the first output (prop) of a previous call to this function.
+        Only measures on properties that are provided both in this dataset and in the properties list will be computed.
+        If None, the second output of the function (meas) will be an empty Dataset.
+    unit_conversion: dict
+        Dictionary of unit conversion to apply to ds before computing the properties.
+        It is useful to be able to convert units, because, for the measure, sim and ref need to have similar units.
```

Comment:
They may not have the same units!
https://github.com/Ouranosinc/xclim/blob/942c69ae40ac5e370bf1571b44898fb8b567fb4c/xclim/sdba/measures.py#L53-L54

Aussi, ne serait-ce pas un travail pour `change_units`, que l'utilisateur devrait appeler avant cette nouvelle fonction? `change_units` a le bonus d'être plus complet (rate <-> amount, par exemple). Personnellement, cet argument ne me semble pas nécessaire ici.

Reply:
en fait, j'ai rajouter ça parce que le `convert_units` après le calcul de propriétés de fonctionnaient pas toujours.

Par exemple, pour la variance de précipitation, `convert_units` n'arrivait à faire mm^2 d^-2  vers kg^2 m-4 s-2.
Cette conversion peut-être fait à la main dans le workflow avant d'appeler properties_and_measures, mais j'aimais mieux ne pas avoir à appeler 2 choses.

on pourrait mettre change_units à la place de convert_units.

Reply:
Ah oui c'est vrai! D'accord alors pour l'option  `change_units` que tu as mise.